### PR TITLE
Fix ClusterExternalSecret FailedNamespaces and ProvisionedNamespaces

### PR DESCRIPTION
**Pull request by shuheiktgw**
## Problem Statement

The more detailed context is in [the previous PR](https://github.com/external-secrets/external-secrets/pull/2490), and it fixes ClusterExternalSecret status's failedNamespaces and provisionedNamespaces filed. Thank you for your review!

## Related Issue

N/A

## Proposed Changes

Update the two fields regardless of the size of the array.

## Checklist

- [x] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [x] All commits are signed with `git commit --signoff`
- [x] My changes have reasonable test coverage
- [x] All tests pass with `make test`
- [x] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

File: pkg/controllers/clusterexternalsecret/clusterexternalsecret_controller_test.go [hunk 1]
```diff
@@ -291,6 +292,67 @@ var _ = Describe("ClusterExternalSecret controller", func() {
 				}
 			},
 		}),
+		Entry("Should crate an external secret and if one with the same name has been deleted", testCase{
```
Comment (shuheiktgw) [inline pkg/controllers/clusterexternalsecret/clusterexternalsecret_controller_test.go:295]: Add this test case to test the failed namespace field 🙂

File: pkg/controllers/clusterexternalsecret/clusterexternalsecret_controller.go [hunk 2]
```diff
@@ -243,30 +241,35 @@ func checkForError(getError error, existingES *metav1.PartialObjectMetadata) str
...
 	for namespace, message := range failedNamespaces {
-		ces.Status.FailedNamespaces = append(ces.Status.FailedNamespaces, esv1beta1.ClusterExternalSecretNamespaceFailure{
+		namespaceFailures[i] = esv1beta1.ClusterExternalSecretNamespaceFailure{
 			Namespace: namespace,
 			Reason:    message,
-		})
+		}
+		i++
 	}
+	sort.Slice(namespaceFailures, func(i, j int) bool { return namespaceFailures[i].Namespace < namespaceFailures[j].Namespace })
```
Comment (shuheiktgw) [inline pkg/controllers/clusterexternalsecret/clusterexternalsecret_controller.go:271]: I believe the namespaceFailures should be sorted too as the provisionedNamespaces are.

File: pkg/controllers/clusterexternalsecret/clusterexternalsecret_controller.go [hunk 3]
```diff
@@ -243,30 +241,35 @@ func checkForError(getError error, existingES *metav1.PartialObjectMetadata) str
...
+	return removedNamespaces
 }
 
-func setFailedNamespaces(ces *esv1beta1.ClusterExternalSecret, failedNamespaces map[string]string) {
-	if len(failedNamespaces) == 0 {
-		return
-	}
-
-	ces.Status.FailedNamespaces = []esv1beta1.ClusterExternalSecretNamespaceFailure{}
+func toNamespaceFailures(failedNamespaces map[string]string) []esv1beta1.ClusterExternalSecretNamespaceFailure {
```
Comment (shuheiktgw) [inline pkg/controllers/clusterexternalsecret/clusterexternalsecret_controller.go:260]: Slightly updated the function to remove the side effect for readability.

**Review by moolen — APPROVED** [review]
Thanks, that improves CES usability by a lot :100:, LGTM!

**Comment by shuheiktgw** [comment]
Thank you for your review!

**Pull request merged**